dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/services/QaService.kt

```diff
@@ -0,0 +1,57 @@
+package org.dataland.datalandqaservice.services
+
+import org.dataland.datalandmessagequeueutils.cloudevents.CloudEventMessageHandler
+import org.dataland.datalandmessagequeueutils.constants.MqConstants
+import org.dataland.datalandmessagequeueutils.exceptions.MessageQueueException
+import org.slf4j.LoggerFactory
+import org.springframework.amqp.core.Message
+import org.springframework.amqp.rabbit.annotation.Exchange
+import org.springframework.amqp.rabbit.annotation.Queue
+import org.springframework.amqp.rabbit.annotation.QueueBinding
+import org.springframework.amqp.rabbit.annotation.RabbitListener
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+/**
+ * Implementation of a QA Service reacting on the upload_queue and forwarding message to qa_queue
+ * @param cloudEventMessageHandler service for managing CloudEvents messages
+ */
+@Component
+class QaService(
+    @Autowired var cloudEventMessageHandler: CloudEventMessageHandler,
+) {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * Method to retrieve message from upload_queue and constructing new one for qa_queue
+     * @param message Message retrieved from upload_queue
+     */
+    @RabbitListener(
+        bindings = [
+            QueueBinding(
+                value = Queue("dataStoredQaService"),
+                exchange = Exchange(MqConstants.dataStored, declare = "false"),
+                key = [""],
+            ),
+        ],
+    )
+    fun assureQualityOfData(message: Message) {
+        val dataId = cloudEventMessageHandler.bodyToString(message)
+        val correlationId = message.messageProperties.headers["cloudEvents:id"].toString()
+        if (dataId.isNotEmpty()) {
+            logger.info(
+                "Received data upload with DataId: $dataId on QA message queue with Correlation Id: $correlationId",
+            )
+            cloudEventMessageHandler.buildCEMessageAndSendToQueue(
+                dataId, "QA Process Completed", correlationId,
```

Comment:
We should also tell the outcome of the QA process in the message

Reply:
Would it be fine for the dummy process if we just write "QA Process Completed - Data Accepted"? During the introduction of the real QA service this would need to change to be dynamic and also the handling of messages accoring to their type has to be introduced to the consumer.

Reply:
sure

Reply:
Done. Services now communicate with Message of Type QaCompletedMessage that includes the dataId and a Qa Result field. As QA service is currently just dummy, there is no reaction to the content of the QA Result implemented yet.

Reply:
ok